src/Entity/Tower.cpp

```diff
@@ -37,22 +37,35 @@ void PlaceNewTower(flecs::iter& iter)
 		return;
 
 	const Vector2 mousePosition = GetMousePosition();
-	const TransformComponent transformComponent{Vector2{mousePosition.x, mousePosition.y}, Vector2{TOWER_WIDTH * ENTITY_SCALE, TOWER_HEIGHT*ENTITY_SCALE}};
+	const TransformComponent transformComponent{Vector2{mousePosition.x, mousePosition.y}, Vector2{TOWER_WIDTH * ENTITY_SCALE, TOWER_HEIGHT * ENTITY_SCALE}};
+	
+	//QUESTION: I am using transform here, but should get the data from collision component. Considering it's always same, should I add global const parameters for it?
+	//what's the best way to "reuse it", considering i will in future have a collider that's different than sprite rect (a sphere on a slime)
+
 	const Rectangle newTowerRect{transformComponent.mPosition.x, transformComponent.mPosition.y, transformComponent.mScale.x, transformComponent.mScale.y};
 
 	bool hasCollided{false};
-
-	//Replaced collision component for now with transform component, same issue as Character.cpp checking on collision on update, cause cant pass two components in query
-	//Assuming at this point, that collision rect is the same as transform rect
-
-	iter.world().each<TransformComponent>([&hasCollided, &newTowerRect](TransformComponent& otherTransformComponent) {
-		// Avoid computing a collision detection if we already detected a collision.
+	Rectangle otherRect{};
+	auto filter = iter.world().filter<CollisionComponent, TransformComponent>();
+	
+	filter.iter([&hasCollided, &newTowerRect, &otherRect](flecs::iter& it, CollisionComponent* collisionComponents, TransformComponent* transformComponents) 
+	{
 		if (hasCollided)
 			return;
+		for (int i = 0; i < it.count(); ++i)
+		{
+		auto collisionComponent = collisionComponents[i];
+		auto transformComponent = transformComponents[i];
+
+		//QUESTION: how do i check that i am not checking the collision "with itself", aka i am checking Object
+		//with all other Objects, so, I am bound to get collision with "itself"?
```

Comment:
In this case, it won't be an issue, the new Tower is not placed yet, so it won't be part of the filter.